modules/packages/Sort.chpl

```diff
@@ -550,13 +556,22 @@ proc isSorted(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator)
 
  */
 iter sorted(x, comparator:?rec=defaultComparator) {
-  var y = x;
-  if !isArrayValue(y) then
-    compilerError("Sort.sorted called on non-iterable");
-
-  sort(y, comparator=comparator);
-  for i in y do
-    yield i;
+  if isArrayValue(x) && Reflection.canResolveMethod(x._value, "dsiSorted", comparator)
+  {
+    for i in x._value.dsiSorted(comparator) {
+      yield i;
+    }
+    return;
```

Comment:
Do we also want to include this check (without the comparator):
```chapel
} else if Reflection.canResolveMethod(_value, "dsiSorted") {
        compilerError(_value.type:string + " does not support dsiSorted(comparator)");
}
```
like the old ChapelArray implementation?